timeserio/preprocessing/datetime.py

```diff
@@ -284,16 +315,76 @@ def transform(self, df):
             )
         return df.reset_index()
 
-    @property
-    def required_columns(self):
-        columns = [self.datetime_column] + _as_list_of_str(self.columns)
-        return set(columns)
 
-    def transformed_columns(self, input_columns):
-        input_columns = set(_as_list_of_str(input_columns))
-        if not self.required_columns <= input_columns:
-            raise ValueError(f'Required columns are {self.required_columns}')
-        lags = _as_list_of_str(self.lags)
-        columns = _as_list_of_str(self.columns)
-        new_columns = [f"{col}_{lag}" for lag in lags for col in columns]
-        return input_columns | set(new_columns)
+class RollingMeanFeaturizer(_BaseLagFeaturizer):
```

Comment:
its probably worth adding this to the `timeserio/preprocessing/__init__.py` so it can be accessed via `preprocessing.RollingMeanFeaturizer` (like `LagFeaturizer` can)